cairo/ethereum/cancun/fork.cairo

```diff
@@ -540,3 +567,151 @@ func process_storage_keys{
     // Process next key
     return process_storage_keys(storage_keys_data, len, index + 1, address);
 }
+
+func check_transaction{
+    range_check_ptr,
+    range_check96_ptr: felt*,
+    add_mod_ptr: ModBuiltin*,
+    mul_mod_ptr: ModBuiltin*,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    state: State,
+}(
+    tx: Transaction,
+    gas_available: Uint,
+    chain_id: U64,
+    base_fee_per_gas: Uint,
+    excess_blob_gas: U64,
+) -> TupleAddressUintTupleVersionedHash {
+    alloc_locals;
+    let gas = TransactionImpl.get_gas(tx);
+    let tx_gas_within_bounds = is_le(gas.value, gas_available.value);
+    with_attr error_message("InvalidBlock") {
+        assert tx_gas_within_bounds = 1;
+    }
+    let sender_address = recover_sender(chain_id, tx);
+    let sender_account = get_account{state=state}(sender_address);
+    let transaction_type = TransactionImpl._get_transaction_type(tx);
+    let is_not_blob_or_fee_transaction = (TransactionType.BLOB - transaction_type) * (
+        TransactionType.FEE_MARKET - transaction_type
+    );
+    // Case where transaction is blob or fee transaction
+    if (is_not_blob_or_fee_transaction == FALSE) {
+        let max_fee_per_gas = TransactionImpl.get_max_fee_per_gas(tx);
+        let max_priority_fee_per_gas = TransactionImpl.get_max_priority_fee_per_gas(tx);
+        let max_fee_per_gas_valid = is_le(base_fee_per_gas.value, max_fee_per_gas.value);
+        let max_priority_fee_per_gas_valid = is_le(
+            max_priority_fee_per_gas.value, max_fee_per_gas.value
+        );
```

Comment:
explain why it is not necessary to use is_le_felt

Reply:
what is the difference?

Reply:
is_le assumes the input is < 2**128 while is-le_felt covers the entire range.

Reply:
then i should probably use is_le_felt here